svn/trunk/SlickEngagement_Plugin.php

```diff
@@ -394,62 +394,76 @@ public function removeSemicolons($value)
     public function addSlickPageHeader()
     {
         global $post;
+
         echo "\n";
-        echo '<meta property="slick:wpversion" content="1.2.4" />' . "\n";
+        echo '<meta property="slick:wpversion" content="1.2.5" />' . "\n";
         $siteCode = trim($this->getOption('SiteCode'));
+
         if ($siteCode) {
             $adThriveAbTest = false;
-            $adThriveAbFraction = 0.9;
-            $serverUrl = trim($this->getOption('SlickServerUrl', 'https://app.slickstream.com'));
+            $defaultServerUrl = 'https://app.slickstream.com';
+            $serverUrl = trim($this->getOption('SlickServerUrl', $defaultServerUrl));
             if (substr($serverUrl, 0, 11) === 'adthrive-ab') {
                 $pieces = explode(" ", $serverUrl);
-                $serverUrl = 'https://app.slickstream.com';
+                $serverUrl = $defaultServerUrl;
                 $adThriveAbTest = true;
-                if (count($pieces) > 1) {
-                    $fractionValue = intval($pieces[1]);
-                    if ($fractionValue > 0 && $fractionValue < 100) {
-                        $adThriveAbFraction = $fractionValue / 100;
-                    }
-                }
+                $enabledPercent = (count($pieces) > 1) ? intval($pieces[1]) : 100;
             }
-            echo '<script>' . "\n";
-            echo '"use strict";' . "\n";
+
+            $jsBlock = <<<JSBLOCK
+            window.slickAbTestResult = function(percentEnabled, forceEnable = false, testName = 'On') {
+                const win = window;
+                const storage = win.localStorage;
+                const targetPercentEnabled = parseInt(percentEnabled);
+                
+                if (isNaN(targetPercentEnabled)) {
+                    return new Error("Invalid enabled percentage");
+                }
+
+                let enableSlickFeature = 1;
+
+                const abGroupVal = `slk\${testName}\${targetPercentEnabled}`;
+                const storedOnOffVal = storage.getItem(abGroupVal);
+                const percentKey = `slickAbTestPercent-\${testName}`;
```

Comment:
can we keep prefix of abGroupVal and percentKey same so it is easier to look up in localStorage

Reply:
I wasn't sure if there was a length limitation in gam, which is why the one that gets sent to the ad code is just `slk`.

Reply:
can we keep abgroup prefix as slk and localStorage prefixes as slick?  they dont need to match and shorter abgroup name is better.  Also, can we update testName to something else? 'On' is confusing, it can be read as test result.

Reply:
The reason the default is just `On` is because it's short for one, but also the full name ends with the percent, so what you end up is something like `slkOn90` which IMO is pretty readable.

Reply:
what is one? that reads like test result is On, which is very confusing for data analysis

Reply:
we can keep it same as before 'slkstm90', or something like 'slkemb90' for embedding slickstream code